src/watson/models.py

```diff
@@ -70,14 +72,21 @@ class SearchEntry(models.Model):
 
     meta_encoded = models.TextField()
 
-    @property
+    def _deserialize_meta(self):
+        from watson.registration import SearchEngine
+        engine = SearchEngine._created_engines[self.engine_slug]
+        model = self.content_type.model_class()
```

Comment:
Without `select_related('content_type')` this will require another query for every item where `.meta` is used.

I've switched to use `cached_property` instead of property so at least accessing `.meta` multiple times won't cause multiple queries.